src/Client.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Http\Client;
+
+/**
+ * Interface for both HTTP Client type
+ *
+ * @author Márk Sági-Kazár mark.sagikazar@gmail.com>
+ */
+interface Client extends PsrHttpClient, HttpClient
```

Comment:
added some comments in #42. now i got what you want to do and it does make sense. and sorry for calling message factory "request factory". i meant indeed message factory.

the discussion about what methods the HttpClient needs should go here so i continue that here :-)
i see the value in having a convenient simple get/post/put method signature. but i would only do that with simply string body and once you need file upload or stream bodies, go to the request factory. although technically this is just syntactic sugar around the message factory - so i guess we can also support the Body interface here to make things more consistent.

ClientTemplate seems a strange name for that trait. can we find a better name? ok, not in this discussion probably.